### PR TITLE
tensorstore 

### PR DESCRIPTION
**Pull request by brokkoli71**
### Description:
- Use tensorstore for zarr arrays
- Reduces the number of IO ops when opening a dataset
- based on changes from #1026

### Todos:
 - [x] Updated Changelog
 - [X] Added / Updated Tests

### CODE REVIEW

File: webknossos/webknossos/dataset/_array.py [hunk 1]
```diff
@@ -296,6 +301,466 @@ def __setstate__(self, d: Dict[str, Any]) -> None:
...
+                "endpoint_url", None
+            ):
+                kvstore_spec["endpoint"] = endpoint_url
+            if "key" in path.storage_options and "secret" in path.storage_options:
+                kvstore_spec["aws_credentials"] = _aws_credential_manager.add(
+                    path.storage_options["key"], path.storage_options["secret"]
+                )
+            return kvstore_spec
+        else:
+            return str(path)
```
Comment (markbader) [inline webknossos/webknossos/dataset/_array.py:386]: Is a string enough for a KVStoreSpec? I would have assumed that something like `{"driver": "file", "path": str(path)}` is required here.

**Review by markbader — COMMENTED** [review]
Looks good already. I just have a few questions and suggested some small changes.

File: webknossos/test.sh [hunk 2]
```diff
@@ -31,6 +31,21 @@ if [ $# -gt 0 ] && [ "$1" = "--refresh-snapshots" ]; then
 
     stop_local_test_wk
 
+    exit $PYTEST_EXIT_CODE
+elif [ $# -gt 0 ] && [ "$1" = "--add-snapshots" ]; then
+    ensure_local_test_wk
+
+    # Starts a proxy server in record mode on port 3000 and sets the HTTP_PROXY env var
+    proxay --mode record --host http://localhost:9000 --tapes-dir tests/cassettes &
```
[the remark this answers is not in the document]
Comment (normanrz) [reply webknossos/test.sh:39]: I usually call the tests I'm interested in directly, e.g. `./test.sh tests/dataset/test_dataset_add_remote_mag_and_layer.py::test_add_remote_mags_from_mag_view`.
Using `mimic` sounds ok too.

File: webknossos/tests/dataset/test_dataset_download_upload_remote.py [hunk 3]
```diff
@@ -90,11 +90,11 @@ def test_remote_dataset(tmp_path: Path) -> None:
...
-    assert remote_ds.display_name is None
-    remote_ds.display_name = "Test Remote Dataset"
-    assert remote_ds.display_name == "Test Remote Dataset"
-    del remote_ds.display_name
-    assert remote_ds.display_name is None
+    # assert remote_ds.display_name is None
+    # remote_ds.display_name = "Test Remote Dataset"
+    # assert remote_ds.display_name == "Test Remote Dataset"
+    # del remote_ds.display_name
+    # assert remote_ds.display_name is None
```
Comment (normanrz) [inline webknossos/tests/dataset/test_dataset_download_upload_remote.py:97]:
```suggestion
    assert remote_ds.display_name is None
    remote_ds.display_name = "Test Remote Dataset"
    assert remote_ds.display_name == "Test Remote Dataset"
    del remote_ds.display_name
    assert remote_ds.display_name is None
```

File: webknossos/tests/test_annotation.py [hunk 4]
```diff
@@ -222,6 +222,7 @@ def check_properties(annotation: wk.Annotation) -> None:
         check_properties(annotation_deserialized)
 
 
+@pytest.mark.skip("issue with current wk version")
```
Comment (normanrz) [inline webknossos/tests/test_annotation.py:225]:
```suggestion
```

File: .github/workflows/ci.yml [hunk 5]
```diff
@@ -189,7 +189,7 @@ jobs:
         timeout-minutes: 30
         env:
           WK_TOKEN: ${{ secrets.WK_TOKEN }}
-        run: ./test.sh -vv -p no:faulthandler --splits 3 --group ${{ matrix.group }} --splitting-algorithm least_duration
+        run: ./test.sh -vv -s -p no:faulthandler --splits 3 --group ${{ matrix.group }} --splitting-algorithm least_duration
```
Comment (normanrz) [inline .github/workflows/ci.yml:192]:
```suggestion
        run: ./test.sh -vv -p no:faulthandler --splits 3 --group ${{ matrix.group }} --splitting-algorithm least_duration
```

**Comment by normanrz** [comment]
I already merged in #1229 here.

File: webknossos/test.sh (hunk 2 above)
[the remark this answers is not in the document]
Comment (normanrz) [reply webknossos/test.sh:39]: `mimic` doesn't work, because it starts replaying if you call the same URL within the same test.

**Review by markbader — APPROVED** [review]
Looks great, let's get it merged :rocket:

**Pull request merged**